docs/apis-tools/frontend-development/03-forms/03-customize-and-extend/02-custom-components.md

```diff
@@ -0,0 +1,40 @@
+---
+id: custom-components
+title: "Custom components"
+description: "Extend form-js with custom components for your domain-specific use cases."
+---
+
+import GHIcon from "@site/src/mdx/GitHubInlineIcon";
+
+Form-js comes with an extension point to hook in custom components. You can define the renderer, the configuration options of the component in the properties panel, and the palette entry.
+
+Custom components are built and distributed separately from the form viewer and renderer, and can be plugged in on demand by registering them as `additionalModules`.
+
+```js
+import { Form } from '@bpmn-io/form-js-viewer';
+import MyCustomComponent from '...';
+
+new Form({
+  container,
+  schema,
+  data,
+  additionalModules: [
+    MyCustomComponent
+  ]
+});
+```
+
+Read the [step-by-step guide](https://github.com/bpmn-io/form-js-examples/tree/master/custom-components) <GHIcon /> and inspect the example component to learn how to write your own custom components.
+
+:::note
+Custom components currently can not be imported into Camunda Web or Desktop Modeler. If you use custom components, you need to host the form editor yourself. Learn more in the [build your own form editor guide](../04-build-your-own-form-editor.md).
```

Comment:
Desktop modeler might be a low hanging fruit though, if you're interested about that I can investigate what's actually required, desktop has a plugin framework it's probably just the matter of writing one helper function or two.

Reply:
Taking a note for that in our team board :)